File: app/src/main/java/com/iita/akilimo/models/CurrentPractice.java
package com.iita.akilimo.models;


import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
@Entity
public class CurrentPractice {

    @Id
    long id;

    private String ploughOperations;
    private String ridgeOperations;
    private String harrowOperations;
    private String weedControlOperations;

    private String ploughingMethod;
    private String ridgingMethod;
    private String harrowingMethod;

    @Getter(AccessLevel.NONE)
    private boolean performPloughing;

    @Getter(AccessLevel.NONE)
    private boolean performHarrowing;

    @Getter(AccessLevel.NONE)
    private boolean performRidging;

    public boolean getPerformPloughing() {
        return this.performPloughing;
    }

    public boolean getPerformHarrowing() {
        return this.performHarrowing;
    }

    public boolean getPerformRidging() {
        return this.performRidging;
    }
}

File: app/src/main/java/com/iita/akilimo/entities/CurrentPractice.kt
package com.iita.akilimo.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "current_practice")
open class CurrentPractice {

    @PrimaryKey(autoGenerate = true)
    var id: Long? = null
    var weedRadioIndex = 0
    var weedControlTechnique: String? = null
    var ploughOperations: String? = null
    var ridgeOperations: String? = null
    var harrowOperations: String? = null
    var weedControlOperations: String? = null
    var ploughingMethod: String? = null
    var ridgingMethod: String? = null
    var harrowingMethod: String? = null
    var tractorAvailable = false
    var tractorPlough = false
    var tractorHarrow = false
    var tractorRidger = false
    var usesHerbicide = false
    var performPloughing = false
    var performHarrowing = false
    var performRidging = false
}
